src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -867,6 +867,36 @@ def build_load_balancer_resource(cmd, name, location, tags, backend_pool_name, n
     return lb
 
 
+def build_nat_rule_v2(cmd, name, location, lb_name, frontend_ip_name, backend_pool_name, backend_port, instance_count,
+                      disable_overprovision):
+    lb_id = "resourceId('Microsoft.Network/loadBalancers', '{}')".format(lb_name)
+
+    nat_rule = {
+        "type": "Microsoft.Network/loadBalancers/inboundNatRules",
+        "apiVersion": get_target_network_api(cmd.cli_ctx),
+        "name": name,
+        "location": location,
+        "properties": {
+            "frontendIPConfiguration": {
+                'id': "[concat({}, '/frontendIPConfigurations/', '{}')]".format(lb_id, frontend_ip_name)
+            },
+            "backendAddressPool": {
+                "id": "[concat({}, '/backendAddressPools/', '{}')]".format(lb_id, backend_pool_name)
+            },
+            "backendPort": backend_port,
+            "frontendPortRangeStart": "50000",
+            "frontendPortRangeEnd": str(max(50119, 49999 + instance_count * (1 if disable_overprovision else 2))),
```

Comment:
Would it be more intuitive to calculate this expression separately and use meaningful variable names?

Reply:
This is a good suggestion. However, unfortunately, this code comes from legacy logic:
https://github.com/Azure/azure-cli/blob/ba4e12556c124cfbdd4d6f6e7409cd2a79f59f97/src/azure-cli/azure/cli/command_modules/vm/_template_builder.py#L827-L828
So I am not very clear about its original meaning. I may add some comments to explain it, what do you think of it?

Reply:
```suggestion
            # This logic comes from the template of `inboundNatPools` to keep consistent with the frontend port of nat pool
            # keep 50119 as minimum for backward compat, and ensure over-provision is taken care of
            "frontendPortRangeEnd": str(max(50119, 49999 + instance_count * (1 if disable_overprovision else 2))),
```

Reply:
Okay, looks great!